servo-media/src/audio/graph.rs

```diff
@@ -90,7 +93,22 @@ impl AudioGraph {
     }
 
     pub fn connect_ports(&self, from: PortId<OutputPort>, to: PortId<InputPort>) {
-        let _ = self.sender.send(AudioRenderThreadMsg::ConnectPorts(from, to));
+        let _ = self.sender
+            .send(AudioRenderThreadMsg::ConnectPorts(from, to));
+    }
+
+    /// Asynchronously decodes the audio file data contained in the given
+    /// buffer.
+    pub fn decode_audio_data(&self, data: Vec<u8>, callbacks: AudioDecoderCallbacks) {
```

Comment:
We should _eventually_ split out the GStreamer stuff as a separate crate with traits, but that's going to be annoying for now so we don't need to do it yet 😄 

(it's also low priority)

Reply:
We already have the `AudioSink` and `AudioDecoder` traits, so it shouldn't be too hard ™ :)